src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -392,10 +393,39 @@ public CloudSdk build() {
         sdkPath = discoverSdkPath();
       }
 
+      // Verify there aren't listeners if subprocess inherits output.
+      // If output is inherited, then listeners won't receive anything.
+      if (inheritProcessOutput
+          && (stdOutLineListeners.size() > 0 || stdErrLineListeners.size() > 0)) {
+        throw new AppEngineException("You cannot specify subprocess output inheritance and"
+            + " output listeners.");
+      }
+
+      // Construct process runner.
+      ProcessRunner processRunner;
+      WaitingProcessOutputLineListener runDevAppServerWaitListener = null;
+      if (stdOutLineListeners.size() > 0 || stdErrLineListeners.size() > 0) {
+        processRunner = new DefaultProcessRunner(async, exitListeners, startListeners,
```

Comment:
This line has to move down to the end of the if block. Otherwise, the runDevAppServerWaitListener never gets added to the DefaultProcessRunner. Notice that the listeners collection is a different ref in DefaultProcessRunner.